core/components/molecules/form/field/field-input.js

```diff
@@ -0,0 +1,116 @@
+import React from 'react'
+import ReactDOM from 'react-dom'
+
+/*
+  There can be just one or multiple input fields inside a Form.Field.
+
+  We want to pass our accessibility props to the first input element.
+  This might be the only child in the field as well.
+
+  However, multiple fields would typically be wrapped inside one
+  layout component like StackLayout which means a single child for us.
+  The layout element should never get the id.
+
+  Here's what we can do:
+
+  1. In render
+
+     If the first child is a cosmos component that renders an input element
+     without a container like ['TextInput', 'TextArea', 'Select'], we can
+     attach an id rightaway.
+
+     If it's one of the cosmos components that has a more complex nested structure
+     like ['Switch', 'Radio', 'Checkbox'] or if the first child is a layout component,
+     we can't attach an id on the parent, we handle this as a progressive enhancement
+     after render.
+
+  2. After render:
+
+     On componentDidMount, we find the first input element and set the id attribute
+     if it wasn't already set.
+
+     This means the rendered output is initially not accessible, but is made
+     accessible progressively.
+*/
+
+const cosmosInputWhitelist = ['TextInput', 'TextArea', 'Select']
```

Comment:
We could use a generic field decorator for this to prevent this list from getting out of sync, so we declare on each input component implementation whether is in the whitelist or not.

+1 on making this progessively